File: CP/Java/Week 3/A_Required_Remainder.java
// https://codeforces.com/problemset/problem/1374/A

import java.util.Scanner;
public class A_Required_Remainder {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int t = scanner.nextInt(); // Number of test cases

        for (int i = 0; i < t; i++) {
            int x = scanner.nextInt();
            int y = scanner.nextInt();
            int n = scanner.nextInt();

            int k = ((n - y) / x) * x + y;

            System.out.println(k);
        }
    }
}

/*
Input
7
7 5 12345
5 0 4
10 5 15
17 8 54321
499999993 9 1000000000
10 5 187
2 0 999999999

Output
12339
0
15
54306
999999995
185
999999998

 */

File: CP/CPP/Week 3/A_Required_Remainder.cpp
// https://codeforces.com/problemset/problem/1374/A

#include<iostream>
using namespace std;

int main()
{
    int t;
    cin>>t;

    while(t--)
    {
    long long int x, y, n;

    cin>>x>>y>>n;

    int a = n%x;
    if(a-y>=0)
        n = n-(a-y);
    else
        n = n-a-x+y;
    cout<<n<<endl;
    }
return 0;
}

/*
input
7
7 5 12345
5 0 4
10 5 15
17 8 54321
499999993 9 1000000000
10 5 187
2 0 999999999

output
12339
0
15
54306
999999995
185
999999998

*/